### ISSUE TITLE
Check organization membership directly?

### ISSUE THREAD

**Issue by kytrinyx** [JamesSingleton/is-organization-member#20]
Hi James,

I'm poking around and am about to use this GitHub Action for a script.

At the moment the script is looping through all organization memberships for a user.

Did you consider using the https://docs.github.com/en/rest/orgs/members#check-organization-membership-for-a-user method instead?

I'm wondering if this would return true for a members that have private memberships, provided that the GitHub token used to make the API call has the right permissions on the organization.

**Comment by JamesSingleton** [comment]
🙈 I honestly didn't think that anyone was using this GitHub Action 😓 I would be open to a PR to support this though!

**Comment by kytrinyx** [comment]
Sweet :-) We ended up taking a different approach.


We ended up with an org-level shared script:
https://github.com/exercism/github-actions/blob/main/.github/workflows/community-contributions.yml

And then use it in a number of repositories here:
https://github.com/exercism/ruby/blob/main/.github/workflows/pause-community-contributions.yml

It was a bit of a hassle, because we discovered (after implementing this) that it didn't report unpublished memberships correctly. Hence the need for the org-level secret that gets passed in to the script:

https://github.com/exercism/ruby/blob/9f873e198a488780e5441b372d2af8baa308397e/.github/workflows/pause-community-contributions.yml#L24C1-L25C85

I'm going to go ahead and close this, since we don't need the change, and if nobody else does either, there's no reason for you to jump through the hoops! 💙 

Thanks for responding (I completely forgot I had opened this)
